### PR TITLE
Installation Markdowns

### PR DESCRIPTION
**Pull request by arunthurai**
## Proposed changes

Describe the changes implemented in this pull request. If the changes fix a bug or resolves a feature request, be sure to link the issue. Please explain the issue and how the changes address the issue.

## Types of changes

What types of changes does your code introduce? Put an `x` in the boxes that apply

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionalitiy)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (if none of the other choices apply)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you are unsure about any of the choices, don't hesitate to ask!

- [ ] Changes have been tested to ensure that fix is effective or that a feature works.
- [ ] Changes pass the unit tests
- [ ] Code has been run through the `poe quality` task
- [ ] I have included necessary documentation or comments (as necessary)
- [ ] Any dependent changes have been merged and published

## Notes
All PRs will undergo the unit testing before being reviewed. You may be requested to explain or make additional changes before the PR is accepted.

### CODE REVIEW

File: docs/contributing/contributing.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# Contributing to Labelmerge
+
+Labelmerge python package dependencies are managed with Poetry (`v1.2.0+`), which
+you will need installed on your machine. You can find installation instructions 
+on the [Poetry website](https://python-poetry.org/docs/master/#installation).
+
+We **strongly** recommend using Labelmerge with the `--use-singularity` flag, which
```
Comment (tkkuehn) [inline docs/contributing/contributing.md:7]: I would soften this a fair bit because if we deploy to PyPI people can just `pip install` this.

File: docs/contributing/contributing.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+the command._
+
+## Dry-run / testing your workflow
+Using Snakemake\'s dry-run option (`--dry-run`/`-n`) is an easy way to verify
+any changes made to the workflow are working direcctly. The `test/data` folder 
+contains a _fake* BIDS dataset (i.e. dataset with zero-sized files) that are 
+useful for verifying different aspects of the workflow. These dry-run tests are 
+part of the automated Github actions that are run for every commit.
+
+You can invoke the pre-configured task via
```
Comment (tkkuehn) [inline docs/contributing/contributing.md:75]: This bit doesn't exist in labelmerge (right now). Should either implement it or get rid of this paragraph.

File: docs/getting_started/docker.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
+# Running Labelmerge with Docker on Windows
```
Comment (tkkuehn) [inline docs/getting_started/docker.md:1]: This is maybe still useful but if we can pip install labelmerge that's probably even easier (at least in some cases)

File: docs/getting_started/docker.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+**Note, these instructions assume you have Docker already installed on a Windows system.
+Docker can also be run on Linux or MacOS with similar commands, but here, we 
+will assume the default Windows CLI is being used.
+
+## First time setup
+
+Open your Windows Command Prompt by clicking the `Windows` button and typing
+`cmd` and pressing the `Enter` on your keyboard. This is where you will enter 
+your SCATTR commands. Feel free to make a new directory with `mkdir` or move to
+a directory you would like to work out of with `cd. For this example, we will
```
Comment (tkkuehn) [inline docs/getting_started/docker.md:12]: Missing closing backtick after `cd

File: docs/getting_started/singularity.md [hunk 5]
```diff
@@ -53,7 +53,7 @@ singularity run -e khanlab_scattr_latest.sif --help-snakemake
 
 ## Running an example
 
-We will use the `test` folder found from the 
+** Provide test example for labelmerge - We will use the `test` folder found from the
```
Comment (tkkuehn) [inline docs/getting_started/singularity.md:56]: Just a note that we should actually do this before merging this PR

File: docs/getting_started/singularity.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
+# Running Labelmerge with Singularity
+
+## Pre-requisites
+1. Singularity / Apptainer is is installed on your system. For more info, see
+the detailed [Apptainer install instructions](https://apptainer.org/docs/admin/main/installation.html#install-from-pre-built-packages).
+1. The following command-line tools are installed:
+    * wget
+1. Sufficient disk-space (rough estimate)
```
Comment (tkkuehn) [inline docs/getting_started/singularity.md:8]: Can soften these requirements a lot (probably just get rid of this list item)

File: docs/getting_started/singularity.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    * in your `/tmp` folder (>30GB) to build the container
+    * in your working folder to store the container (~20GB)
+    * for SCATTR outputs (~40GB per subject using default parameters)
+1. Sufficient CPU and memory - the more you have, the faster it will run and
```
Comment (tkkuehn) [inline docs/getting_started/singularity.md:12]: Probably don't need 8 cores and 64GB memory for an example Labelmerge

File: docs/getting_started/singularity.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+```
+
+## Running an example
+
+** Provide test example for labelmerge - We will use the `test` folder found from the 
+[Github repository](https://github.com/khanlab/labelmerge/tree/main/test/) to
+demonstrate an example of how to run SCATTR:
+
+```
+singularity run -e khanlab_scattr_latest.sif test/data/bids test/data/derivatives participant --fs-license test/fs_license --force-output -n
```
Comment (tkkuehn) [inline docs/getting_started/singularity.md:61]: This CLI invocation is for SCATTR, so update to Labelmerge

File: docs/getting_started/singularity.md [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+the rules that will be run. We can also have a shell command used for each rule
+printed to screen using the `-p` Snakemake option
+
+```
+singularity run -e khanlab_labelmerge_latest.sif test/data/bids test/data/derivatives participant --fs-license test/fs_license --force-output -np
+```
+
+Now to actually run the workflow, we need to specify how many cores to use and 
+leave out the dry-run option. The Snakemake `--cores` option tells Labelmerge how
+many cores to use. Using `--cores 8` means that SCATTR will only make use of 8
```
Comment (tkkuehn) [inline docs/getting_started/singularity.md:91]: SCATTR -> Labelmerge

File: docs/usage/cli.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
+# Command Line Interface (CLI)
+
+## labelmerge CLI
+The following can also be seen by entering `labelmerge -h` into your terminal.
+
+These are all the required and optional arguments labelmerge accepts in order to 
+run flexibly on many different input data types and with many options. In most 
+cases, only the required arguments are neede.
```
Comment (tkkuehn) [inline docs/usage/cli.md:8]: `neede` -> needed

File: docs/getting_started/installation.md [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+
+BIDS App for combining multiple atlas parcellations
+
+## Requirements
+
+* Docker (Intel Mac/Windows/Linux) or Singularity (Linux)
+* For those wishing to contribute or modify the code, `pip install` or `poetry
+install` are also available (Linux), but will still require Singularity to 
+handle some dependencies. See [Contributing to Labelmerge](https://labelmerge.readthedocs.io/en/stable/contributing/contributing.html).
+* _Note: Apple ARM-based chips (e.g. M1, M2, etc.) are **not currently
```
Comment (tkkuehn) [inline docs/getting_started/installation.md:11]: I don't think this is true for Labelmerge

File: docs/getting_started/installation.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
+# Installation
+
+BIDS App for combining multiple atlas parcellations
+
+## Requirements
+
+* Docker (Intel Mac/Windows/Linux) or Singularity (Linux)
```
Comment (tkkuehn) [inline docs/getting_started/installation.md:7]: Or pip? We should decide if we want to release this via PyPI

File: docs/getting_started/docker.md (hunk 3 above)
Comment (kaitj) [reply docs/getting_started/docker.md:1]: I agree though I would leave the container stuff in to give people different options

File: docs/usage/cli.md [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+func: get_parser
+prog: labelmerge
+---
+```
+
+## Snakemake CLI
+In addition to the above command line arguments, Snakemake arguments can also be
+passed at the labelmerge command line.
+
+The most critical of these is the `--cores / -c` and `--force-output` arguments,
```
[the remark this answers is not in the document]
Comment (kaitj) [reply docs/usage/cli.md:22]: It shouldn’t be for labelmerge. It was in scattr because there are rules that write to existing derivative folders as a result of using freesurfer

File: docs/getting_started/docker.md [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+Pull the container (this will take some time and storage stage, but like an 
+installation, it only needs to be done once and can be then be run on many 
+datasets). The example below pulls the latest versioned container (replace 
+`latest` with `vX.X.X` for a specific version).
+
+```
+docker pull khanlab/labelmerge:latest
+```
+
+Run without any arguments to print hte short help:
```
Comment (kaitj) [inline docs/getting_started/docker.md:28]: Hte -> the

File: docs/getting_started/docker.md [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+container upon completion (`--rm`) and mount the the directoty 
+(`-v c:\Users\username\Downloads\labelmerge\test`) to a directory within the
+container named `\test`. These are not specific to Labelmerge, but are general ways
+to use Docker. You may want to familiarize yourself with 
+[Docker options](https://docs.docker.com/engine/reference/run/).
+
+The first three arguments to Labelmerge (as with any BIDS App) are the input folder 
+(`test/data/bids`), the output folder (`test/data/derivatives`), and the 
+analysis level (`participant`). The `participant` analysis level is used in 
+SCATTR to perform further participant-level processing of Freesurfer (thalamus
```
Comment (kaitj) [inline docs/getting_started/docker.md:72]: Scattr related documentation, can be deleted

File: docs/getting_started/installation.md [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+
+The Labelmerge BIDS App is available on DockerHub as versioned releases.
+Instructions can be found in the [Docker](https://labelmerge.readthedocs.io/en/stable/getting_started/docker.html) documentation page.
+
+### Pros
+* Compatible with non-Linux systems
+* All dependencies are in a single container
+
+### Cons
+* Typically not possible on shared macines
```
Comment (kaitj) [inline docs/getting_started/installation.md:27]: Macines -> machines

File: docs/getting_started/singularity.md (hunk 7 above)
Comment (kaitj) [reply docs/getting_started/singularity.md:12]: I don’t think we need more than 8gb maybe 16g for labelmerge

File: docs/workflow/simple_workflow.png
Comment (kaitj) [inline docs/workflow/simple_workflow.png:1]: I would modify this image for the repository to just show what labelmerge does. It’s not clear what the “original application” is referring to here.

**Review by kaitj — CHANGES_REQUESTED** [review]

File: docs/contributing/contributing.md [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+
+Labelmerge python package dependencies are managed with Poetry (`v1.2.0+`), which
+you will need installed on your machine. You can find installation instructions 
+on the [Poetry website](https://python-poetry.org/docs/master/#installation).
+
+_Note: These instructions are only recommended if you are making changes to the
+Labelmerge codebase and committing these back to the repository, or if you are 
+using Snakemake's cluster execution profiles. If not, it is easier to run 
+Labelmerge using the packaged singularity container (e.g. 
+`docker://khanlab/labelmerge:latest`)._
```
Comment (kaitj) [inline docs/contributing/contributing.md:11]: Would recommend using a version-tagged container if possible

File: docs/contributing/contributing.md [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+git clone https://github.com/khanlab/labelmerge.git 
+cd labelmerge
+poetry install --with dev 
+```
+
+Poetry will automatically create a virtual environment. To customize where 
+these virtual environments are stored, see the poetry docs 
+[here](https://python-poetry.org/docs/configuration/)
+
+Then, you can run labelmerge with:
```
Comment (kaitj) [inline docs/contributing/contributing.md:27]: This feels like this should be a separate part of the doc on how to run labelmerge rather than contribution.

File: docs/contributing/contributing.md [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+```
+poetry shell
+labelmerge
+```
+
+You can exit the poetry shell with `exit`
+
+## Running and fixing code format quality
+
+labelmerge uses [poethepoet](https://github.com/nat-n/poethepoet) as a task runner.
```
Comment (kaitj) [inline docs/contributing/contributing.md:44]: labelmerge -> Labelmerge

File: docs/getting_started/docker.md [hunk 20]
```diff
@@ -0,0 +1,78 @@
...
+docker run -it --rm khanlab/labelmerge:latest
+```
+
+Use the `-h` option to get a detailed help listing:
+
+```
+docker run -it --rm khanlab_labelmerge_latest.sif -h
+```
+
+_Note that all the Snakemake command-line options are also available in SCATTR,
```
Comment (kaitj) [inline docs/getting_started/docker.md:40]: This is referencing scattr

File: docs/getting_started/docker.md [hunk 21]
```diff
@@ -0,0 +1,78 @@
...
+and can be listed with `--help-snakemake`:
+
+```
+docker run -it --rm khanlab_labelmerge_latest.sif --help-snakemake
+```
+
+### Explanation
+
+Everything prior to the container (`khanlab_labelmerge_latest.sif`) are arguments
+to Docker and after are to Lablemerge itself. The first three arguments to Docker
```
Comment (kaitj) [inline docs/getting_started/docker.md:50]: Spelling of labelmerge

File: docs/getting_started/singularity.md [hunk 22]
```diff
@@ -0,0 +1,92 @@
...
+Run Labelmerge without any arguments to print the short help:
+
+```
+singularity run -e khanlab_labelmerge_latest.sif
+```
+
+Use the `-h` option to get a detailed help listing:
+
+```
+singularity run -e khanlab_scattr_latest.sif -h
```
Comment (kaitj) [inline docs/getting_started/singularity.md:39]: scattr reference

File: docs/getting_started/singularity.md [hunk 23]
```diff
@@ -0,0 +1,92 @@
...
+```
+
+_Note that all the Snakemake command-line options are also available in SCATTR,
+and can be listed with `--help-snakemake`:
+
+```
+singularity run -e khanlab_scattr_latest.sif --help-snakemake
```
Comment (kaitj) [inline docs/getting_started/singularity.md:46]: scattr reference

File: docs/getting_started/singularity.md [hunk 24]
```diff
@@ -0,0 +1,92 @@
...
+```
+
+## Running an example
+
+** Provide test example for labelmerge - We will use the `test` folder found from the 
+[Github repository](https://github.com/khanlab/labelmerge/tree/main/test/) to
+demonstrate an example of how to run SCATTR:
```
Comment (kaitj) [inline docs/getting_started/singularity.md:53]: scattr reference

File: docs/getting_started/singularity.md [hunk 25]
```diff
@@ -0,0 +1,92 @@
...
+
+```
+singularity run -e khanlab_labelmerge_latest.sif test/data/bids test/data/derivatives participant
+```
+
+### Explanation
+
+Everything prior to the container (`khanlab_labelmerge_latest.sif`) are arguments
+to Singluarity / Apptainer, and after are to Labelmerge itself. The first three arguments to 
+Lablemerge (as with any BIDS App) are the input folder (`test/data/bids`), the
```
Comment (kaitj) [inline docs/getting_started/singularity.md:63]: Labelmerge spelling

File: docs/usage/cli.md [hunk 26]
```diff
@@ -0,0 +1,35 @@
...
+## Snakemake CLI
+In addition to the above command line arguments, Snakemake arguments can also be
+passed at the labelmerge command line.
+
+
+The complete list of [Snakemake](https://snakemake.readthedocs.io/en/stable/) 
+arguments are below, and most act to determine your environment and app
+behaviours. They will likely only need to be used for running in cloud
+environments or troubleshooting. These can be listed from the command line with
+`scatter --help-snakemake`.
```
Comment (kaitj) [inline docs/usage/cli.md:27]: scattr reference

File: docs/usage/faq.md [hunk 27]
```diff
@@ -0,0 +1,20 @@
...
+
+The workflow was unable to find the required input files from the input BIDS
+directory. This can happen if:
+
+* Singularity or Docker cannot access your input directory. For Singularity,
+ensure your 
+[Singularity options](https://docs.sylabs.io/guides/3.1/user-guide/cli/singularity_run.html) 
+are appropriate, in particular `SINGULARITY_BINDPATH`. For Docker, ensure you 
+are mounting the correct directory with the `-v` flag described in the 
+[Getting Started](https://scattr.readthedocs.io/en/stable/getting_started/docker.html)
```
Comment (kaitj) [inline docs/usage/faq.md:13]: Points to scattr docs

File: docs/workflow/workflow.md [hunk 28]
```diff
@@ -0,0 +1,30 @@
+# Workflow Details
+
+This section describes the Labelmerge workflow (i.e. steps taken to produce 
+intermediate and final files). Labelmerge is a 
+[Snakemake](https://snakemake.readthedocs.io/en/stable/) workflow, and thus a 
+directed acyclic graph (DAG) that is automatically configured based on a set of 
+rules. An simplified workflow schematic is also shown below.
```
Comment (kaitj) [inline docs/workflow/workflow.md:7]: "An simplified" -> "A simplified"

**Review by kaitj — CHANGES_REQUESTED** [review]
There are still a few references to scattr, as well as misspelling of labelmerge. Would also try to be consistent with capitalization (labelmerge vs Labelmerge), which is used interchangeably throughout.

File: docs/workflow/workflow.md [hunk 29]
```diff
@@ -0,0 +1,30 @@
...
+in the workflow source. 
+
+At a more granular level, Labelmerge reads one “base” and one “overlay” atlas image organized as BIDS 
+derivatives, with the label names supplied as metadata in separate TSV files. 
+An image mask is created for both the base and overlay images, mapping each unique 
+label name in the parcellation scheme to their respective voxel coordinates. The two masks 
+are then merged, with labels reindexed to range from one to the total number of labels across both images. 
+An output image is constructed where any voxel that is labeled in both the base and overlay images takes 
+the reindexed value from the overlay image. Finally, this image is written alongside a new metadata table 
+where the label names are pre-appended with “base “ or “overlay “ as appropriate.
```
Comment (kaitj) [inline docs/workflow/workflow.md:30]: "pre-appended" -> "prepend"

File: docs/workflow/workflow.md [hunk 30]
```diff
@@ -0,0 +1,30 @@
...
+also exist in separate rule files, which can be found under the 
+[rules sub-directories](https://github.com/khanlab/labelmerge/tree/main/labelmerge/workflow/rules) 
+in the workflow source. 
+
+At a more granular level, Labelmerge reads one “base” and one “overlay” atlas image organized as BIDS 
+derivatives, with the label names supplied as metadata in separate TSV files. 
+An image mask is created for both the base and overlay images, mapping each unique 
+label name in the parcellation scheme to their respective voxel coordinates. The two masks 
+are then merged, with labels reindexed to range from one to the total number of labels across both images. 
+An output image is constructed where any voxel that is labeled in both the base and overlay images takes
```
Comment (kaitj) [inline docs/workflow/workflow.md:28]: Is there a way to reword this to make it more clear - wording slightly confusing.

**Review by kaitj — CHANGES_REQUESTED** [review]

File: docs/workflow/workflow.md [hunk 31]
```diff
@@ -0,0 +1,28 @@
...
+_(Click on the image to enlarge)_
+
+Snakemake workflows are organized into groups of rules, each
+representing the different phases of the workflow. Each grouped set of rules 
+also exist in separate rule files, which can be found under the 
+[rules sub-directories](https://github.com/khanlab/labelmerge/tree/main/labelmerge/workflow/rules) 
+in the workflow source. 
+
+At a more granular level, Labelmerge reads one “base” and one “overlay” atlas image organized as BIDS 
+derivatives, with the label names supplied as metadata in separate TSV files. Brain Masks are data dictionaries,
```
Comment (kaitj) [inline docs/workflow/workflow.md:24]: If you can, would provide a link to the BEP that has the specs. 

Also "Brain Masks" -> "brain masks"

File: docs/workflow/workflow.md [hunk 32]
```diff
@@ -0,0 +1,28 @@
...
+Snakemake workflows are organized into groups of rules, each
+representing the different phases of the workflow. Each grouped set of rules 
+also exist in separate rule files, which can be found under the 
+[rules sub-directories](https://github.com/khanlab/labelmerge/tree/main/labelmerge/workflow/rules) 
+in the workflow source. 
+
+At a more granular level, Labelmerge reads one “base” and one “overlay” atlas image organized as BIDS 
+derivatives, with the label names supplied as metadata in separate TSV files. Brain Masks are data dictionaries, 
+mapping each label to the associated spatial region. The merge labels step is achieved by updating the 
+empty mask in order of the overlay, exceptions, and base masks, overwriting any duplicate labels.
```
Comment (kaitj) [inline docs/workflow/workflow.md:26]: This should be arranged accordingly since you mention its done "in order" and "overwrites". I believe its something like base -> overlay -> exceptions. Or would rephrase to make more clear.

File: docs/getting_started/singularity.md [hunk 33]
```diff
@@ -0,0 +1,109 @@
...
+to already existing folders - in this case, for thalamus segmentations via 
+Freesurfer. We also used the `--dry-run/-n` option to print out what would run,
+without actually running the workflow.
+
+When you run the above command, a long listing will print out, describing all 
+the rules that will be run. We can also have a shell command used for each rule
+printed to screen using the `-p` Snakemake option
+
+```
+singularity run -e khanlab_labelmerge_latest.sif test/data/bids test/data/derivatives participant --fs-license test/fs_license --force-output -np
```
[the remark this answers is not in the document]
Comment (kaitj) [reply docs/getting_started/singularity.md:59]: fs-license is still mentioned here.

File: docs/contributing/contributing.md [hunk 34]
```diff
@@ -0,0 +1,60 @@
+# Contributing to Labelmerge
+
+Labelmerge python package dependencies are managed with Poetry (`v1.2.0+`), which
+you will need installed on your machine. You can find installation instructions 
+on the [Poetry website](https://python-poetry.org/docs/master/#installation).
+
+_Note: These instructions are only recommended if you are making changes to the
+Labelmerge codebase and committing these back to the repository, or if you are 
+using Snakemake's cluster execution profiles. If not, it is easier to run 
+Labelmerge using the packaged singularity container (e.g.
```
Comment (kaitj) [inline docs/contributing/contributing.md:10]: I think we can omit the word "singularity" here.

File: docs/contributing/contributing.md [hunk 35]
```diff
@@ -0,0 +1,60 @@
...
+git clone https://github.com/khanlab/labelmerge.git 
+cd labelmerge
+poetry install --with dev 
+```
+
+Poetry will automatically create a virtual environment. To customize where 
+these virtual environments are stored, see the poetry docs 
+[here](https://python-poetry.org/docs/configuration/)
+
+Then, you can run labelmerge with:
```
Comment (kaitj) [inline docs/contributing/contributing.md:25]: Also need to capitalize "labelmerge" I think for consistency.

File: docs/contributing/contributing.md [hunk 36]
```diff
@@ -0,0 +1,60 @@
...
+## Running and fixing code format quality
+
+Labelmerge uses [poethepoet](https://github.com/nat-n/poethepoet) as a task runner.
+You can see what commands are available by running:
+
+```
+poetry run poe 
+```
+
+We use a a few tools, including `black`, `flake8`, `isort`, `snakefmt`, and
```
Comment (kaitj) [inline docs/contributing/contributing.md:49]: `flake8` is going to change to `ruff` in an upcoming PR. I can make the change to the docs once this is merged in, otherwise I wouldn't mind if you changed it here either.

File: docs/getting_started/docker.md [hunk 37]
```diff
@@ -0,0 +1,78 @@
+# Running Labelmerge with Docker on Windows
+
+**Note, these instructions assume you have Docker already installed on a Windows system.
```
Comment (kaitj) [inline docs/getting_started/docker.md:3]: Hmm, not sure what formatting this was meant to be - maybe italicized? I don't think the way `**` is used here will actually result in any sort of formatting and will be rendered as is.

File: docs/getting_started/docker.md [hunk 38]
```diff
@@ -0,0 +1,78 @@
...
+your commands. Feel free to make a new directory with `mkdir` or move to
+a directory you would like to work out of with `cd'. For this example, we will
+work from:
+
+```
+cd c:\Users\username\Downloads
+```
+
+Pull the container (this will take some time and storage stage, but like an 
+installation, it only needs to be done once and can be then be run on many
```
Comment (kaitj) [inline docs/getting_started/docker.md:20]: Reword -> "can be then be run"

File: docs/getting_started/installation.md [hunk 39]
```diff
@@ -0,0 +1,55 @@
...
+
+Instructions can be found in the [Contributing](https://labelmerge.readthedocs.io/en/stable/contributing/contributing.html) documentation page.
+
+### Pros
+* Flexibility to modify code
+
+### Cons
+* Only compatible on systems with Singularity for external dependencies
+
+## Pip Install
```
Comment (kaitj) [inline docs/getting_started/installation.md:49]: I think the requirements should be updated to include python if are including pip install instructions. Would also include pros / cons similar to the others if possible.

File: docs/getting_started/installation.md [hunk 40]
```diff
@@ -0,0 +1,55 @@
+# Installation
+
+BIDS App for combining multiple atlas parcellations
+
+## Requirements
+
+* Docker (Intel Mac/Windows/Linux) or Singularity (Linux)
+
+### Notes
+* Inputs to Labelmerge should be organized as BIDS derivatives, taking one "base" and one "overlay" atlas
```
Comment (kaitj) [inline docs/getting_started/installation.md:10]: Should provide a link to the BIDS derivatives spec so that users can see an example.

**Review by kaitj — CHANGES_REQUESTED** [review]

File: docs/usage/cli.md [hunk 41]
```diff
@@ -0,0 +1,35 @@
+# Command Line Interface (CLI)
+
+## labelmerge CLI
```
Comment (kaitj) [inline docs/usage/cli.md:3]: labelmerge -> Labelmerge

File: docs/usage/cli.md [hunk 42]
```diff
@@ -0,0 +1,35 @@
...
+The following can also be seen by entering `labelmerge -h` into your terminal.
+
+These are all the required and optional arguments labelmerge accepts in order to
```
Comment (kaitj) [inline docs/usage/cli.md:6]: labelmerge -> Labelmerge

File: docs/usage/cli.md [hunk 43]
```diff
@@ -0,0 +1,35 @@
...
+---
+filename: ../labelmerge/run.py
+func: get_parser
+prog: labelmerge
+---
+```
+
+## Snakemake CLI
+In addition to the above command line arguments, Snakemake arguments can also be
+passed at the labelmerge command line.
```
Comment (kaitj) [inline docs/usage/cli.md:20]: labelmerge -> Labelmerge

File: docs/requirements.txt
Comment (arunthurai) [inline docs/requirements.txt:1]: include python3.7 here as a requirement? and would we also need pip then

Comment (kaitj) [reply docs/requirements.txt:1]: This `requirements.txt` is solely for the documentation and isn't actually part of the package (see `pyproject.toml` for those dependencies.

File: docs/getting_started/singularity.md (hunk 6 above)
Comment (tkkuehn) [reply docs/getting_started/singularity.md:8]: This comment still stands -- I don't think people need anything like 16GB memory to run this

File: docs/usage/cli.md [hunk 44]
```diff
@@ -0,0 +1,35 @@
+# Command Line Interface (CLI)
```
Comment (tkkuehn) [inline docs/usage/cli.md:1]: The autogenerated arguments as it renders on readthedocs don't seem to include the arguments defined in the config. Those are really important, and I think it's worth going through them explicitly anyway. (i.e. `--overlay_bids_dir`, `--base_desc`, `--overlay_desc`, and the exceptions and drops args)

File: docs/usage/cli.md [hunk 45]
```diff
@@ -0,0 +1,35 @@
...
+
+## Labelmerge CLI
+The following can also be seen by entering `labelmerge -h` into your terminal.
+
+These are all the required and optional arguments Labelmerge accepts in order to 
+run flexibly on many different input data types and with many options. In most
```
Comment (tkkuehn) [inline docs/usage/cli.md:7]: I don't even really think the last bit is true -- you're almost always going to need either `--overlay_bids dir` or `--base_desc` and `--overlay_desc`

**Review by tkkuehn — CHANGES_REQUESTED** [review]
Almost there, just a few suggestions that I think will help these docs be maximally useful

File: docs/usage/cli.md (hunk 44 above)
Comment (arunthurai) [reply docs/usage/cli.md:1]: Otherwise can manually write those for here

File: docs/usage/cli.md (hunk 45 above)
Comment (arunthurai) [reply docs/usage/cli.md:7]: changed to include thse

File: docs/getting_started/installation.md [hunk 46]
```diff
@@ -0,0 +1,60 @@
+# Installation
+
+BIDS App for combining multiple atlas parcellations
+
+## Requirements
+
+* Docker (Intel Mac/Windows/Linux) or Singularity (Linux)
+
+### Notes
+* Inputs to Labelmerge should be organized as [BIDS derivatives](https://bids-specification.readthedocs.io/en/stable/05-derivatives/03-imaging.html#segmentations), taking one "base" and one "overlay" atlas
```
[the remark this answers is not in the document]
Comment (arunthurai) [reply docs/getting_started/installation.md:8]: added to cli section

File: docs/usage/cli.md (hunk 44 above)
Comment (kaitj) [reply docs/usage/cli.md:1]: Yeah, you will need to add the `add_dynamic_args` in if it isn't already there.

**Review by tkkuehn — COMMENTED** [review]
Need to run `black` on this for the docs to build, but the changes are looking good

File: labelmerge/run.py [hunk 47]
```diff
@@ -8,6 +8,7 @@
 def get_parser():
     """Exposes parser for sphinx doc generation, cwd is the docs dir"""
     app = SnakeBidsApp("../labelmerge", skip_parse_args=True)
+    add_dynamic_args(app.parser, app.config["parse_args"], app.config["pybids_inputs"])
```
Comment (tkkuehn) [inline labelmerge/run.py:11]: Need to add `from snakebids.cli import add_dynamic_args` above with the other imports for this to work

**Comment by arunthurai** [comment]
Checks look like they pass

**Review by kaitj — APPROVED** [review]
